test/fgpu/test_main.py

```diff
@@ -0,0 +1,127 @@
+        with pytest.raises(ValueError, match="invalid literal for int"):
+            assert comma_split(int)("3,hello")
+
+    def test_fixed_count(self) -> None:
+        """Test with a value for `count`."""
+        splitter = comma_split(int, 2)
+        assert splitter("3,5") == [3, 5]
+        with pytest.raises(ValueError, match="Expected 2 comma-separated fields, received 3"):
+            splitter("3,5,7")
+        with pytest.raises(ValueError, match="Expected 2 comma-separated fields, received 1"):
+            splitter("3")
+
+    def test_allow_single(self) -> None:
+        """Test with `allow_single`."""
+        splitter = comma_split(int, 2, allow_single=True)
+        assert splitter("3,5") == [3, 5]
+        assert splitter("3") == [3, 3]
+        with pytest.raises(ValueError, match="Expected 2 comma-separated fields, received 3"):
+            splitter("3,5,7")
+
+
+class TestParseNarrowband:
+    """Test :func:`.parse_narrowband`."""
+
+    def test_minimal(self) -> None:
+        """Test with the minimum required arguments."""
+        assert parse_narrowband("channels=1024,decimation=8,dst=239.1.2.3+1:7148") == {
+            "channels": 1024,
+            "decimation": 8,
+            "dst": [Endpoint("239.1.2.3", 7148), Endpoint("239.1.2.4", 7148)],
+        }
+
+    def test_maximal(self) -> None:
+        """Test with all valid arguments."""
+        assert parse_narrowband("channels=1024,decimation=8,taps=8,w_cutoff=0.5,dst=239.1.2.3+1:7148") == {
+            "channels": 1024,
+            "decimation": 8,
+            "taps": 8,
+            "w_cutoff": 0.5,
+            "dst": [Endpoint("239.1.2.3", 7148), Endpoint("239.1.2.4", 7148)],
+        }
+
+    @pytest.mark.parametrize(
+        "missing,value",
+        [
+            ("channels", "decimation=8,dst=239.1.2.3+1:7148"),
+            ("decimation", "channels=1024,dst=239.1.2.3+1:7148"),
+            ("dst", "channels=1024,decimation=8"),
+        ],
+    )
+    def test_missing_key(self, missing: str, value: str) -> None:
+        """Test without one of the required keys."""
+        with pytest.raises(ValueError, match=f"--narrowband: {missing} is missing"):
+            parse_narrowband(value)
+
+    def test_duplicate_key(self) -> None:
+        """Test with a key specified twice."""
+        with pytest.raises(ValueError, match="--narrowband: channels specified twice"):
+            parse_narrowband("channels=8,channels=9,decimation=8,dst=239.1.2.3+1:7148")
+
+
+class TestParseArgs:
+    """Test :func:`.katgpucbf.fgpu.main.parse_args`."""
+
+    def test_narrowband_defaults(self) -> None:
+        """Test that missing narrowband config is taken from the global config."""
+        raw_args = [
+            "--src-interface=lo",
+            "--dst-interface=lo",
+            "--adc-sample-rate=1712000000.0",
+            "--channels=1024",
+            "--sync-epoch=0",
+            "--taps=64",
+            "--w-cutoff=0.9",
```

Comment:
The Engine __init__() method refactor was pretty good. I almost want to suggest we add a 
```
--wideband=...
```
command-line option as well, because it looks a bit more consistent.

BUT

that would break backwards compatibility so I'm not sure whether it would be worth the effort.

Reply:
It's definitely something I'm considering adding (in a future PR), particularly since I said I wanted to try adding support for multiple widebands (to flush out all the issues with having multiple streams) before trying to support narrowband.

Backwards compatibility can be finessed e.g. fall back to the current behaviour if zero `--wideband` options are given.